Cython/Debugger/libcython.py

```diff
@@ -165,6 +226,45 @@ def __init__(self,
 
 # General purpose classes
 
+def frame_repr(frame):
+    """Returns a string representing the internal state of a provided GDB frame
+
+    Created to serve as GDB.Frame.__repr__ for debugging purposes. GDB has many
+    layers of abstraction separating the state of the debugger from the
+    corresponding source code. This prints a tree of instance properties,
+    expanding the values for Symtab_and_line, Symbol, and Symtab.
+
+    Most of these properties require computation to determine, meaning much of
+    relevant info is behind a monad. Rather than list which properties are worth
+    printing, we assume that frames are mostly a read-only context and that any
+    function without an argument is idempotent. That is to say, every function
+    is called without arguments and any that don't throw an exception are added
+    to the output.
+
+    Arguments
+    frame       The GDB.Frame instance to be represented as a string
+    """
+    res = str(frame) + "\n"
+    for attribute in sorted(dir(frame)):
+        if attribute.startswith("__"):
+            continue
+        value = getattr(frame, attribute)
+        if callable(value):
+            try:
+                value = value()
+            except Exception as e:
+                pass
+
+        if type(value) in [gdb.Symtab_and_line, gdb.Symbol, gdb.Symtab]:
+            # strip last line since it will get added on at the end of the loop
+            value = frame_repr(value).strip("\n").replace("\n", "\n\t")
+        res += attribute + ": "
+        if type(value) == int:
```

Comment:
This is an unusual spelling. You'd normally expect `is` for a type check like this rather than `==`, but why not just use the normal `isinstance()`?
```suggestion
        if isinstance(value, int):
```